### PR TITLE
Complete user friendships functionality

### PR DESCRIPTION
**Pull request by johnktravers**
Buttons appear next to github followers and followings that have accounts in the system. Once clicked, a friendship is established. All edge cases are tested. Test coverage is at 87%.

### CODE REVIEW

File: app/controllers/friendships_controller.rb [hunk 1]
```diff
@@ -2,13 +2,14 @@ class FriendshipsController < ApplicationController
...
     friendship = current_user.friendships.new(friend_id: params[:friend_id])
-    new_friend = User.find(friendship.friend_id)
+    new_friend = User.find_by(id: friendship.friend_id)
+
     if new_friend && friendship.save
       flash[:success] = "Yay, you added #{new_friend.first_name} #{new_friend.last_name} as a friend!"
+      redirect_to dashboard_path
     else
-      flash[:notice] = "That user doesn't exist"
+      render file: 'public/404', status: :not_found
```
Comment (Jonpatt92) [inline app/controllers/friendships_controller.rb:11]: Nice sad path handling

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -14,26 +14,23 @@
...
-    <ul>
-      <% current_user.all_friends.each do |friend| %>
-        <li>
-          <%= friend.github_username %>
-        </li>
-      <% end %>
-    </ul>
-
-  </section>
+  <% if current_user.friends.any? %>
```
Comment (Jonpatt92) [inline app/views/users/show.html.erb:22]: Good catch

**Review by Jonpatt92 — COMMENTED** [review]
Looks great, we're almost wrapped up. I'll grab the champagne
